main.go

```diff
@@ -0,0 +1,388 @@
+		if err != nil {
+			return err
+		}
+		if baseDesc.Platform == nil {
+			return fmt.Errorf("unknown platform for image: %v", baseRef)
+		}
+		if err := verifyPlatform(*baseDesc.Platform); err != nil {
+			return err
+		}
+		logf := withPrefix(logf, fmt.Sprintf("%v/%v: ", baseDesc.Platform.OS, baseDesc.Platform.Architecture))
+		img, err := createImageForBase(logf, staticFiles, gopaths, ldFlags, baseImage)
+		if err != nil {
+			return err
+		}
+		if !push {
+			logf("not pushing")
+			return nil
+		}
+
+		for _, r := range refs {
+			logf("pushing to %v", r)
+			if err := remote.Write(r, img, remoteOpts...); err != nil {
+				return err
+			}
+		}
+		return nil
+	case types.OCIImageIndex, types.DockerManifestList:
+		// baseRef is a multi-platform index, rest of the method handles this.
+	default:
+		return fmt.Errorf("failed to interpret base as index or image: %v", baseDesc.MediaType)
+	}
+	baseIndex, err := baseDesc.ImageIndex()
+	if err != nil {
+		return err
+	}
+
+	im, err := baseIndex.IndexManifest()
+	if err != nil {
+		return fmt.Errorf("failed to interpret base as index: %w", err)
+	}
+	var adds []mutate.IndexAddendum
+	// Try to build images for all supported platforms.
+	for _, id := range im.Manifests {
+		logf := withPrefix(logf, fmt.Sprintf("%v/%v: ", id.Platform.OS, id.Platform.Architecture))
+		if err := verifyPlatform(*id.Platform); err != nil {
+			logf("skipping: %v", err)
+			continue
+		}
+		logf("base digest: %v", id.Digest)
+		bi, err := baseIndex.Image(id.Digest)
+		if err != nil {
+			return err
+		}
+		logf("building")
+		img, err := createImageForBase(logf, staticFiles, gopaths, ldFlags, bi)
+		if err != nil {
+			return err
+		}
+		d, err := img.Digest()
+		if err != nil {
+			return err
+		}
+		logf("new digest: %v", d)
+		adds = append(adds, mutate.IndexAddendum{
+			Add: img,
+			Descriptor: v1.Descriptor{
+				MediaType:   id.MediaType,
+				URLs:        id.URLs,
+				Annotations: id.Annotations,
+				Platform:    id.Platform,
+			},
+		})
+	}
+
+	// Generate a new index with all the platform images.
+	idx := mutate.AppendManifests(mutate.IndexMediaType(empty.Index, types.OCIImageIndex), adds...)
+	d, err := idx.Digest()
+	if err != nil {
+		return err
+	}
+	logf("index digest: %v", d)
+	if !push {
+		logf("not pushing")
+		return nil
+	}
+
+	for _, r := range refs {
+		logf("pushing to %v", r)
+		if err := remote.WriteIndex(r, idx, remoteOpts...); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func createImageForBase(logf logf, staticFiles, goPaths map[string]string, ldFlags string, base v1.Image) (v1.Image, error) {
+	cf, err := base.ConfigFile()
+	if err != nil {
+		return nil, err
+	}
+	tmpDir, err := os.MkdirTemp("", "mkctr")
+	if err != nil {
+		return nil, err
+	}
+	defer os.RemoveAll(tmpDir)
+
+	env := append(os.Environ(),
+		"CGO_ENABLED=0",
+		"GOOS="+cf.OS,
+		"GOARCH="+cf.Architecture,
+	)
+	if cf.Architecture == "arm" {
+		env = append(env, "GOARM=5")
```

Comment:
for all ARM or just the old ARM?

I don't know what values cf.Architecture can have.

But we don't want GOARM=5 for e.g. new Raspberry Pis.

Reply:
parsed the arch varient